<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1119</width>
    <height>898</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>190</y>
      <width>1121</width>
      <height>21</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_2">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>440</y>
      <width>1131</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_3">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>450</y>
      <width>20</width>
      <height>441</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="QTextBrowser" name="school_display">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>40</y>
      <width>261</width>
      <height>131</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="school_query">
    <property name="geometry">
     <rect>
      <x>196</x>
      <y>17</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>院系查询</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>54</width>
      <height>12</height>
     </rect>
    </property>
    <property name="text">
     <string>学校管理</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>210</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>院系管理</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="teacher_display">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>316</y>
      <width>181</width>
      <height>131</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="teacher_query">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>290</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>老师查询</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="school_id">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>230</y>
      <width>131</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>240</y>
      <width>91</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>操作院系编号</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="add_teacher_id">
    <property name="geometry">
     <rect>
      <x>183</x>
      <y>316</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>300</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>教师编号</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="add_teacher_name">
    <property name="geometry">
     <rect>
      <x>183</x>
      <y>364</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>348</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>教师姓名</string>
    </property>
   </widget>
   <widget class="QPushButton" name="add_teacher">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>410</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>添加老师</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="student_display">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>316</y>
      <width>181</width>
      <height>131</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="add_student_id">
    <property name="geometry">
     <rect>
      <x>473</x>
      <y>295</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>279</y>
      <width>51</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>学号</string>
    </property>
   </widget>
   <widget class="QPushButton" name="add_student">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>420</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>添加学生</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="add_student_name">
    <property name="geometry">
     <rect>
      <x>473</x>
      <y>343</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="student_queery">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>292</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>学生查询</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>327</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>学生姓名</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>780</x>
      <y>327</y>
      <width>51</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>班号</string>
    </property>
   </widget>
   <widget class="QPushButton" name="add_class">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>387</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>添加班级</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="class_display">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>316</y>
      <width>181</width>
      <height>131</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="add_class_id">
    <property name="geometry">
     <rect>
      <x>753</x>
      <y>341</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="class_query">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>290</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>班级查询</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="classroom_id">
    <property name="geometry">
     <rect>
      <x>1031</x>
      <y>316</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="classroom_addr">
    <property name="geometry">
     <rect>
      <x>1031</x>
      <y>364</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>1048</x>
      <y>348</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>教室地址</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="classroom_display">
    <property name="geometry">
     <rect>
      <x>850</x>
      <y>316</y>
      <width>181</width>
      <height>131</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="classroom_query">
    <property name="geometry">
     <rect>
      <x>850</x>
      <y>290</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>教室查询</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>1048</x>
      <y>300</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>教室编号</string>
    </property>
   </widget>
   <widget class="QPushButton" name="add_classroom">
    <property name="geometry">
     <rect>
      <x>1038</x>
      <y>410</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>添加教室</string>
    </property>
   </widget>
   <widget class="QPushButton" name="del_teacher">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>290</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>删除老师</string>
    </property>
   </widget>
   <widget class="QPushButton" name="del_student">
    <property name="geometry">
     <rect>
      <x>397</x>
      <y>292</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>删除学生</string>
    </property>
   </widget>
   <widget class="QPushButton" name="del_class">
    <property name="geometry">
     <rect>
      <x>673</x>
      <y>290</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>删除班级</string>
    </property>
   </widget>
   <widget class="QPushButton" name="del_classroom">
    <property name="geometry">
     <rect>
      <x>956</x>
      <y>290</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>删除教室</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="course_display">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>558</y>
      <width>181</width>
      <height>131</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>452</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>教师课程管理</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="add_course_id">
    <property name="geometry">
     <rect>
      <x>183</x>
      <y>558</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="del_course">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>532</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>删除</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_14">
    <property name="geometry">
     <rect>
      <x>207</x>
      <y>542</y>
      <width>51</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>课程号</string>
    </property>
   </widget>
   <widget class="QPushButton" name="add_course">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>700</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>添加课程</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_15">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>482</y>
      <width>91</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>操作教师编号</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="teacher_id">
    <property name="geometry">
     <rect>
      <x>92</x>
      <y>482</y>
      <width>131</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="add_course_name">
    <property name="geometry">
     <rect>
      <x>183</x>
      <y>606</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="course_query">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>532</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>查询</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_16">
    <property name="geometry">
     <rect>
      <x>208</x>
      <y>590</y>
      <width>51</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>课程名</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_17">
    <property name="geometry">
     <rect>
      <x>204</x>
      <y>640</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>占用教室</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="add_course_classroom">
    <property name="geometry">
     <rect>
      <x>183</x>
      <y>656</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="free_cr_query">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>532</y>
      <width>111</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>可用教室查询</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="free_classroom_display_2">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>558</y>
      <width>181</width>
      <height>131</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_18">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>450</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>学生事务管理</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="student_id">
    <property name="geometry">
     <rect>
      <x>774</x>
      <y>460</y>
      <width>131</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_19">
    <property name="geometry">
     <rect>
      <x>684</x>
      <y>460</y>
      <width>91</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>操作学生学号</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="add_school_id">
    <property name="geometry">
     <rect>
      <x>273</x>
      <y>46</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_20">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>30</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>院系编号</string>
    </property>
   </widget>
   <widget class="QPushButton" name="add_school">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>140</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>添加院系</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="add_school_name">
    <property name="geometry">
     <rect>
      <x>273</x>
      <y>94</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_21">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>78</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>院系名称</string>
    </property>
   </widget>
   <widget class="QPushButton" name="free_course_query">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>504</y>
      <width>101</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>可选课程查询</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="free_course_display">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>530</y>
      <width>181</width>
      <height>131</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="selected_course_query">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>502</y>
      <width>81</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>已选课程查询</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="selected_course_display">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>528</y>
      <width>181</width>
      <height>131</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="drop_course">
    <property name="geometry">
     <rect>
      <x>663</x>
      <y>502</y>
      <width>81</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>退课</string>
    </property>
   </widget>
   <widget class="QPushButton" name="select_course">
    <property name="geometry">
     <rect>
      <x>1027</x>
      <y>504</y>
      <width>81</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>选课</string>
    </property>
   </widget>
   <widget class="QPushButton" name="borrow_book">
    <property name="geometry">
     <rect>
      <x>1027</x>
      <y>686</y>
      <width>81</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>借书</string>
    </property>
   </widget>
   <widget class="QPushButton" name="borrowed_book_query">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>684</y>
      <width>81</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>已借图书查询</string>
    </property>
   </widget>
   <widget class="QPushButton" name="free_book_query">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>686</y>
      <width>101</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>可借图书查询</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="free_book_display">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>712</y>
      <width>181</width>
      <height>131</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="return_book">
    <property name="geometry">
     <rect>
      <x>663</x>
      <y>684</y>
      <width>81</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>还书</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="borrowed_book_display">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>710</y>
      <width>181</width>
      <height>131</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextBrowser" name="lib_display">
    <property name="geometry">
     <rect>
      <x>379</x>
      <y>37</y>
      <width>261</width>
      <height>131</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="lib_query">
    <property name="geometry">
     <rect>
      <x>382</x>
      <y>13</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>图书馆查询</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="books_desplay">
    <property name="geometry">
     <rect>
      <x>677</x>
      <y>38</y>
      <width>261</width>
      <height>131</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="book_query">
    <property name="geometry">
     <rect>
      <x>861</x>
      <y>13</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>藏书查询</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="book_query_lib_id">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>5</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_22">
    <property name="geometry">
     <rect>
      <x>690</x>
      <y>10</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>图书馆编号</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_23">
    <property name="geometry">
     <rect>
      <x>987</x>
      <y>84</y>
      <width>51</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>图书名称</string>
    </property>
   </widget>
   <widget class="QPushButton" name="add_book">
    <property name="geometry">
     <rect>
      <x>977</x>
      <y>146</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>添加图书</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_24">
    <property name="geometry">
     <rect>
      <x>987</x>
      <y>36</y>
      <width>51</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>图书编号</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="add_book_id">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>52</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="add_book_name">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>100</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="del_book">
    <property name="geometry">
     <rect>
      <x>860</x>
      <y>170</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>删除图书</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="add_student_name_2">
    <property name="geometry">
     <rect>
      <x>473</x>
      <y>390</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_25">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>374</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>所在班级</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="student_id_2">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>600</y>
      <width>131</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="student_id_3">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>790</y>
      <width>131</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_26">
    <property name="geometry">
     <rect>
      <x>810</x>
      <y>570</y>
      <width>91</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>课程号</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_27">
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>760</y>
      <width>91</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>图书编号</string>
    </property>
   </widget>
   <widget class="QPushButton" name="create_view">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>170</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>生成视图</string>
    </property>
   </widget>
   <widget class="QPushButton" name="having_test">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>800</y>
      <width>121</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>having_test</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1119</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
